employees/common/exports.py

```diff
@@ -8,217 +6,212 @@
-        worksheet,
-        current_row,
-        ExcelGeneratorSettingsConstants.HOURS_COLUMN_FOR_SINGLE_USER.value,
-        ExcelGeneratorSettingsConstants.TOTAL_HOURS_FORMULA_FOR_SINGLE_USER.value,
-    )
-    return workbook
+    return ReportExtractor().generate_xlsx_for_single_user(author)
 
 
 def generate_xlsx_for_project(project: Project) -> Workbook:
-    authors = project.members.order_by("-last_name")
-    workbook = Workbook()
-    for author in authors:
-        worksheet = set_active_worksheet_name(workbook, author)
-        reports = project.report_set.filter(author=author.pk).order_by("-date")
+    return ReportExtractor().generate_xlsx_for_project(project)
+
+
+class ReportExtractor:
+    def __init__(self):
+        self._current_row = -1
+        self._workbook = None
+        self._headers = []
+        self._description_column_index = -1
+        self._hours_column_index = -1
+        self._daily_hours_column_index = -1
+        self._formula = ""
+        self._last_date = None
+        self._active_worksheet = None
+        self._headers_settings = {}
+
+    def generate_xlsx_for_project(self, project: Project) -> Workbook:
+        authors = project.members.all()
+        self._workbook = Workbook()
+        self._set_xlsx_settings_for_project_report()
+        for author in authors:
+            employee_name = get_employee_name(author)
+            reports = project.report_set.filter(author=author.pk).order_by("date")
+            if not reports:
+                continue
+
+            self._fill_report_for_single_user(employee_name, reports)
+
+        self._workbook._sheets.sort(key=lambda w: str.lower(w.title))
+        return self._workbook
+
+    def generate_xlsx_for_single_user(self, author: CustomUser) -> Workbook:
+        reports = author.get_reports_created().order_by("date", "project__name")
+        self._workbook = Workbook()
+        self._set_xlsx_settings_for_user_report()
         employee_name = get_employee_name(author)
-        fill_headers(
-            worksheet,
-            ExcelGeneratorSettingsConstants.HEADERS_FOR_USER_IN_PROJECT.value,
-            ExcelGeneratorSettingsConstants.COLUMNS_WIDTH_FOR_PROJECT.value,
-            employee_name,
-        )
 
-        current_row = ExcelGeneratorSettingsConstants.FIRST_ROW_FOR_DATA.value
-        reports_date = []
+        self._fill_report_for_single_user(employee_name, reports)
+        return self._workbook
+
+    def _fill_report_for_single_user(self, employee_name, reports):
+        self._prepare_worksheet(employee_name)
         for report in reports:
-            (reports_date, report_date, daily_hours) = get_report_date_and_daily_hours(
-                reports_date, report.date, reports
+            self._fill_single_report(report, reports)
+
+        self._summarize_user_reports()
+
+    def _fill_single_report(self, report, reports):
+        report_date, daily_hours = self._get_report_date_and_daily_hours(report, reports)
```

Comment:
I personally prefer to add tuple as a return when I have more than one return element, because I had once problem with this kind of case and I learned that this is like a little safety check :D